### PR TITLE
Validate all of RT! Aggregated to the hour per feed, combined with RT parsing

### PR DESCRIPTION
**Pull request by atvaccaro**
# Description

This PR combines RT parsing and validation into a common framework to minimize the amount of network traffic and output a single combined error file covering the feed-hour.

See https://console.cloud.google.com/storage/browser/test-rt-parsed for some example output, for all three categories (actual data, validations, and errors).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation
- [ ] agencies.yml

## How has this been tested?

Running locally, testing via local Airflow soon.

## Screenshots (optional)
```
writing 166 lines to test-rt-parsed/vehicle_positions/dt=2022-04-22/itp_id=98/url_number=0/hour=12/gtfs_rt_vehicle_positions_url.jsonl.gz
writing 199 lines to test-rt-parsed/vehicle_positions_validations/dt=2022-04-22/itp_id=98/url_number=0/hour=12/gtfs_rt_vehicle_positions_url.jsonl.gz
WARNING: no records at all for test-rt-parsed/vehicle_positions_errors/dt=2022-04-22/itp_id=98/url_number=0/hour=12/gtfs_rt_vehicle_positions_url
validating /var/folders/ls/cdr1pq6n7yn9m0cb41zkflk80000gn/T/tmpim_bahn7/rt_62c420a40dbfbab984038b3ee235872a/ with /var/folders/ls/cdr1pq6n7yn9m0cb41zkflk80000gn/T/tmpim_bahn7/gtfs_62c420a40dbfbab984038b3ee235872a.zip
executing rt_validator: java -jar /Users/andrewvaccaro/rt-validator.jar -gtfs /var/folders/ls/cdr1pq6n7yn9m0cb41zkflk80000gn/T/tmpim_bahn7/gtfs_62c420a40dbfbab984038b3ee235872a.zip -gtfsRealtimePath /var/folders/ls/cdr1pq6n7yn9m0cb41zkflk80000gn/T/tmpim_bahn7/rt_62c420a40dbfbab984038b3ee235872a/ -sort name
writing 305 lines to test-rt-parsed/vehicle_positions/dt=2022-04-22/itp_id=99/url_number=0/hour=12/gtfs_rt_vehicle_positions_url.jsonl.gz
WARNING: no records at all for test-rt-parsed/vehicle_positions_validations/dt=2022-04-22/itp_id=99/url_number=0/hour=12/gtfs_rt_vehicle_positions_url
WARNING: no records at all for test-rt-parsed/vehicle_positions_errors/dt=2022-04-22/itp_id=99/url_number=0/hour=12/gtfs_rt_vehicle_positions_url
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 78/78 [06:00<00:00,  4.62s/it]
fin.
```

### CODE REVIEW

**Comment by atvaccaro** [comment]
I think I may have introduced a memory leak.

**Review by lauriemerrell — COMMENTED** [review]
1. Seems like parse job is missing for service alerts?
2. Wondering if we want to be able to pass a parameter to suppress parsing or validation? like if we want to bump validator version and backfill validations only, do we want to be able to do that without re-parsing? and same question if we change/fix parsing logic, can we re-run that without re-validating everything?

File: jobs/gtfs-rt-parser/LICENSE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
[the remark this answers is not in the document]
Comment (atvaccaro) [reply jobs/gtfs-rt-parser/LICENSE:1]: This comes from the RT validator itself, Michael included it in the other repo.

**Comment by atvaccaro** [comment]
> 1. Seems like parse job is missing for service alerts?+

There should be one, `parse_rt_service_alerts`.

> 2. Wondering if we want to be able to pass a parameter to suppress parsing or validation? like if we want to bump validator version and backfill validations only, do we want to be able to do that without re-parsing? and same question if we change/fix parsing logic, can we re-run that without re-validating everything?

Airflow does support parameterized runs, or we can handle it in code. It's hard to design it generally but also allow us to more efficiently backfill by combining the jobs (specifically the raw reads).

File: jobs/gtfs-rt-parser/README.md [hunk 2]
```diff
@@ -5,6 +5,10 @@ via BigQuery external tables. This is primarily so we can run this code as a Pod
 a special jobs node pool; if we ever self-host Airflow and use the [KubernetesExecutor](https://airflow.apache.org/docs/apache-airflow/stable/executor/kubernetes.html), this
 separate image will be redundant.
 
+# gtfs-rt-validator-api
+
+This is basically a ground-up refactor of https://github.com/cal-itp/gtfs-rt-validator-api.
```
[the remark this answers is not in the document]
Comment (atvaccaro) [reply jobs/gtfs-rt-parser/README.md:10]: This specific one, yes. Since we really are just wrapping the "actual" project i.e. the Java validator.

**Comment by lauriemerrell** [comment]
IDK what I'm missing but I don't see service alerts:
image

File: jobs/gtfs-rt-parser/Dockerfile [hunk 3]
```diff
@@ -1,11 +1,20 @@
-FROM python:3.7-buster
+FROM openjdk:11
 
 LABEL org.opencontainers.image.source https://github.com/cal-itp/data-infra
 
-RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+ENV GTFS_RT_VALIDATOR_JAR=/gtfs-realtime-validator.jar
+ENV GTFS_VALIDATOR_VERSION=v1.0.0
```
[the remark this answers is not in the document]
Comment (atvaccaro) [reply jobs/gtfs-rt-parser/Dockerfile:6]: I'm open to taking a "cut" like you have suggested in the past.

This code here is straight from https://github.com/cal-itp/gtfs-rt-validator-api/blob/main/Dockerfile#L15-L17

File: jobs/gtfs-rt-parser/gtfs_rt_parser.py [hunk 4]
```diff
@@ -111,92 +216,259 @@ def identify_files(glob, rt_file_type: RTFileType, progress=False) -> List[RTFil
...
 
+def download_gtfs_schedule_zip(gtfs_schedule_path, dst_path, pbar=None):
+    # fetch and zip gtfs schedule
+    zipname = f"{dst_path[:-1]}.zip"
+    log(
+        f"Fetching gtfs schedule data from {gtfs_schedule_path} to {zipname}",
+        pbar=pbar,
+    )
+
+    files = [
```
[the remark this answers is not in the document]
Comment (atvaccaro) [reply jobs/gtfs-rt-parser/gtfs_rt_parser.py:233]: Oh I looked at a list of files/tables somewhere (tbh I don't remember exactly), I guess we get them in some feeds?

I can remove if needed.

**Comment by atvaccaro** [comment]
> IDK what I'm missing but I don't see service alerts: image

Oh ok you're looking at the files changed tab? The only changes I made to the actual parsing YAML in _this_ PR was changing resource requests on vehicle positions and trip updates.

**Comment by atvaccaro** [comment]
I'm going to merge into the [OG PR ](https://github.com/cal-itp/data-infra/pull/1458) and we can discuss there

**Pull request merged**